### PR TITLE
[search] [search_quality] A refactoring of the search quality lib and tools.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: search/search_quality/helpers.cpp [hunk 1]
```diff
@@ -1,28 +1,65 @@
...
+#include "base/logging.hpp"
 #include "base/string_utils.hpp"
 
 #include "std/target_os.hpp"
 
+#include <fstream>
+#include <limits>
 #include <memory>
 #include <utility>
+#include <vector>
```
Comment (tatiana-yan) [inline search/search_quality/helpers.cpp]: есть в hpp

Comment (mpimenov) [reply search/search_quality/helpers.cpp]: done

File: search/search_quality/search_quality_tool/search_quality_tool.cpp [hunk 2]
```diff
@@ -68,17 +68,69 @@ DEFINE_string(viewport, "", "Viewport to use when searching (default, moscow, lo
...
+  string part;
+  while (getline(iss, part, delim))
+    parts.push_back(part);
+}
+
 struct CompletenessQuery
 {
+  DECLARE_EXCEPTION(MalformedQueryException, RootException);
+
+  explicit CompletenessQuery(string s)
```
Comment (tatiana-yan) [inline search/search_quality/search_quality_tool/search_quality_tool.cpp]: string && s ?

File: search/search_quality/search_quality_tool/search_quality_tool.cpp [hunk 3]
```diff
@@ -68,17 +68,69 @@ DEFINE_string(viewport, "", "Viewport to use when searching (default, moscow, lo
...
+
+    vector<string> parts;
+    Split(s, ';', parts);
+    if (parts.size() != 7)
+    {
+      MYTHROW(MalformedQueryException,
+              ("Can't split", s, ", found", parts.size(), "part(s):", parts));
+    }
+
+    auto idx = parts[0].find(':');
```
Comment (tatiana-yan) [inline search/search_quality/search_quality_tool/search_quality_tool.cpp]: const?

File: search/search_quality/helpers.cpp [hunk 4]
```diff
@@ -1,28 +1,63 @@
...
 using namespace std;
 
 namespace
 {
-void ParsePoint(json_t * root, m2::PointD & point)
+uint64_t ReadVersionFromHeader(platform::LocalCountryFile const & mwm)
 {
-  FromJSONObject(root, "x", point.x);
-  FromJSONObject(root, "y", point.y);
+  vector<string> specialFiles = {WORLD_FILE_NAME, WORLD_COASTS_FILE_NAME,
```
Comment (tatiana-yan) [inline search/search_quality/helpers.cpp]: const?

Comment (mpimenov) [reply search/search_quality/helpers.cpp]: done

File: search/search_quality/search_quality_tool/search_quality_tool.cpp (hunk 3 above)
Comment (mpimenov) [reply search/search_quality/search_quality_tool/search_quality_tool.cpp]: done

File: search/search_quality/search_quality_tool/search_quality_tool.cpp (hunk 2 above)
Comment (mpimenov) [reply search/search_quality/search_quality_tool/search_quality_tool.cpp]: done

**Comment by tatiana-yan** [comment]
JTMAC

**Comment by tatiana-yan** [comment]
JTALL

**Pull request merged**